plugins/out_s3/s3.c

```diff
@@ -1033,7 +1033,7 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
     }
 
     len = strlen(s3_key);
-    if ((len + 16) <= 1024) {
+    if ((len + 16) <= 1024 && !strstr(ctx->s3_key_format, "$UUID")) {
         append_random = FLB_TRUE;
```

Comment:
Checking the key format for every call is inefficient and unnecessary since the format doesn't change. You can move this check to the plugin init function (`cb_s3_init`) and then store the flag `append_random` on the context (may be choose a more descriptive name for it on the context).

Reply:
I think we still need to keep `append_random` here because we need to create `uri` by different lengths. It depends on how long the `s3_key` would be and if `s3_key_format` has `$UUID`.

Reply:
But I would improve the code by moving the checking to init function